loopchain/peer/consensus_siever.py

```diff
@@ -162,8 +173,7 @@ def __build_candidate_block(self, block_builder, next_leader, vote_result):
             if await self._wait_for_voting(candidate_block) is None:
                 return
 
-            if len(candidate_block.body.transactions) == 0 and not conf.ALLOW_MAKE_EMPTY_BLOCK and \
-                    next_leader.hex_hx() != ChannelProperty().peer_id:
+            if self.made_block_count >= (conf.MAX_MADE_BLOCK_COUNT - 1) and next_leader.hex_hx() != ChannelProperty().peer_id:
```

Comment:
This condition "self.made_block_count >= (conf.MAX_MADE_BLOCK_COUNT - 1)" forces not to change leader even if next_leader is changed. But block has already been made with the "next_leader".

It may not be a problem at the moment, but it is likely to cause problems later.